cmd/vic-machine/common/utils.go

```diff
@@ -57,8 +59,30 @@ func CheckUnsupportedCharsDatastore(s string) error {
 func checkUnsupportedChars(s string, re *regexp.Regexp) error {
 	st := []byte(s)
 	var v []int
+	//this is validation step for characters in a datastore URI
 	if v = re.FindIndex(st); v == nil {
 		return nil
 	}
 	return fmt.Errorf("unsupported character %q in %q", s[v[0]:v[1]], s)
 }
+
+// CheckNFSURlValidation is used to validate propernfs arguments to the --volume-store flag and avoid checking inputs for datastore restricted characters if a url is provided
+func CheckURLValidation(input string) *url.URL {
+	parts := strings.Split(input, ":")
```

Comment:
Yes, it should since this is the initial check on the input. The input for --volume-store is <volume store target>`:`<label>. Because of this parts must always be atleast 2 or more. In which case it should likely be greater than 1, whoops.... :/ but we can assume a lower bound of the parts for sure.